app/policies/component_policy.rb

```diff
@@ -1,12 +1,34 @@
 class ComponentPolicy < ApplicationPolicy
-  attr_reader :user, :component
+  attr_reader :user, :component, :msa_component, :sp_component
 
   def initialize(user, component)
     @user = user
     @component = component
+    @msa_component = component
+    @sp_component = component
```

Comment:
how come we need all three of these and not just the msa and sp ones?

Reply:
Was thinking the same is ```@component = component``` needed

Reply:
This is because the policy is inherited by two other policies (msa_component_policy and sp_component_policy).